### PR TITLE
publish gems to github; add badges to README.md

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Voxpupuli Test Gem
 
+[![License](https://img.shields.io/github/license/voxpupuli/voxpupui-test.svg)](https://github.com/voxpupuli/voxpupui-test/blob/master/LICENSE)
+[![Test](https://github.com/voxpupuli/voxpupui-test/actions/workflows/test.yml/badge.svg)](https://github.com/voxpupuli/voxpupui-test/actions/workflows/test.yml)
+[![codecov](https://codecov.io/gh/voxpupuli/voxpupui-test/branch/master/graph/badge.svg?token=Mypkl78hvK)](https://codecov.io/gh/voxpupuli/voxpupui-test)
+[![Release](https://github.com/voxpupuli/voxpupui-test/actions/workflows/release.yml/badge.svg)](https://github.com/voxpupuli/voxpupui-test/actions/workflows/release.yml)
+[![RubyGem Version](https://img.shields.io/gem/v/voxpupui-test.svg)](https://rubygems.org/gems/voxpupui-test)
+[![RubyGem Downloads](https://img.shields.io/gem/dt/voxpupui-test.svg)](https://rubygems.org/gems/voxpupui-test)
```
Comment (ekohl) [inline README.md:8]: voxpupui -> voxpupuli

File: voxpupuli-test.gemspec [hunk 2]
```diff
@@ -45,4 +45,6 @@ Gem::Specification.new do |s|
 
   # development
   s.add_development_dependency 'rspec'
+  s.add_development_dependency 'simplecov-console'
+  s.add_development_dependency 'codecov'
```
Comment (ekohl) [inline voxpupuli-test.gemspec:49]: Can we make this a group in `Gemfile` and only include conditionally?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -13,17 +13,23 @@ jobs:
...
+            coverage: "yes"
     steps:
       - uses: actions/checkout@v2
       - name: Install Ruby ${{ matrix.ruby }}
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
+        env:
+          BUNDLE_WITHOUT: release
```
Comment (ekohl) [inline .github/workflows/test.yml:31]: Why is this needed? On line 7/8 it's already set.

**Review by ekohl — CHANGES_REQUESTED** [review]

**Review by ekohl — APPROVED** [review]

**Review by llowder — APPROVED** [review]

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1 +1,24 @@
...
+    # do not track vendored files
+    add_filter '/vendor'
+    add_filter '/.vendor'
+  end
+
+  SimpleCov.formatters = [
+    SimpleCov::Formatter::Console,
+    SimpleCov::Formatter::Codecov,
+  ]
+end
```
Comment (ekohl) [inline spec/spec_helper.rb:22]:
```suggestion
begin
  require 'simplecov'
  require 'simplecov-console'
  require 'codecov'
rescue LoadError
else
  SimpleCov.start do
    track_files 'lib/**/*.rb'

    add_filter '/spec'

    enable_coverage :branch

    # do not track vendored files
    add_filter '/vendor'
    add_filter '/.vendor'
  end

  SimpleCov.formatters = [
    SimpleCov::Formatter::Console,
    SimpleCov::Formatter::Codecov,
  ]
end
```

File: Gemfile [hunk 5]
```diff
@@ -5,3 +5,8 @@ gemspec
 group :release do
   gem 'github_changelog_generator', :require => false
 end
+
+group :coverage do
```
Comment (ekohl) [inline Gemfile:9]: On IRC I suggested to make it optional:
```suggestion
group :coverage, optional: !ENV['COVERAGE'] do
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -26,4 +28,6 @@ jobs:
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
       - name: Run tests
+        env:
+          COVERAGE: ${{ matrix.coverage }}
```
Comment (ekohl) [inline .github/workflows/test.yml:32]: I think you need to set this at a higher level or at least duplicate it for the installation step

**Review by sebastianrakel — APPROVED** [review]

**Pull request merged**